storage/store.go

```diff
@@ -447,6 +455,9 @@ func (s *Store) configGossipUpdate(key string, contentsChanged bool) {
 func (s *Store) maybeSplitRangesByConfigs(configMap PrefixConfigMap) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
+	if s.splitQueue == nil {
```

Comment:
You then wouldn't need this check either. Just let all this machinery work as is and be defensive in split queue.